library/core/class.dispatcher.php

```diff
@@ -796,6 +797,13 @@ private function _mapParts($Parts, $ControllerKey) {
             }
         }
     }
+
+    /**
+     * Trigger an event allowing addons to modify response headers.
+     */
+    public function sendHeaders() {
```

Comment:
If the sole purpose of this function is to be added as a callback you should do 
```php
$that = $this;
header_register_callback(function() use ($that) {
    $that->fireEvent('SendHeaders');
});
```
(Not tested but should work.)